pkg/apiserver/auth-config_test.go

```diff
@@ -39,158 +41,153 @@ import (
+		Fail("requestheader-group-headers not stored correctly")
 	}
 
 	if !reflect.DeepEqual(deserializeStringSlice(cm.Data["requestheader-extra-headers-prefix"]), authConfig.ExtraPrefixHeaders) {
-		t.Errorf("requestheader-extra-headers-prefix not stored correctly")
+		Fail("requestheader-extra-headers-prefix not stored correctly")
 	}
 }
 
-func TestNewCdiAPIServer(t *testing.T) {
-	ch := make(chan struct{})
-	kubeobjects := []runtime.Object{}
-	kubeobjects = append(kubeobjects, getAPIServerConfigMap(t))
+type fakeCertWatcher struct {
+	cert *tls.Certificate
+}
 
-	client := k8sfake.NewSimpleClientset(kubeobjects...)
-	aggregatorClient := aggregatorapifake.NewSimpleClientset()
-	cdiClient := cdiclientfake.NewSimpleClientset()
-	authorizer := &testAuthorizer{}
-	authConfigWatcher := NewAuthConfigWatcher(client, ch)
+func (fcw *fakeCertWatcher) GetCertificate(_ *tls.ClientHelloInfo) (*tls.Certificate, error) {
+	return fcw.cert, nil
+}
 
-	server, err := NewCdiAPIServer("0.0.0.0", 0, client, aggregatorClient, cdiClient, authorizer, authConfigWatcher, nil)
-	if err != nil {
-		t.Errorf("Upload api server creation failed: %+v", err)
-	}
+func NewFakeCertWatcher() CertWatcher {
+	certBytes, keyBytes, _ := k8scert.GenerateSelfSignedCertKey("foobar", nil, nil)
+	c, _ := tls.X509KeyPair(certBytes, keyBytes)
+	return &fakeCertWatcher{cert: &c}
+}
 
-	app := server.(*cdiAPIApp)
+var _ = Describe("Auth config tests", func() {
+	It("New CDI API server", func() {
+		ch := make(chan struct{})
+		kubeobjects := []runtime.Object{}
+		kubeobjects = append(kubeobjects, getAPIServerConfigMap())
 
-	req, _ := http.NewRequest("GET", "/apis", nil)
-	rr := httptest.NewRecorder()
+		client := k8sfake.NewSimpleClientset(kubeobjects...)
+		aggregatorClient := aggregatorapifake.NewSimpleClientset()
+		cdiClient := cdiclientfake.NewSimpleClientset()
+		authorizer := &testAuthorizer{}
+		authConfigWatcher := NewAuthConfigWatcher(client, ch)
 
-	app.container.ServeHTTP(rr, req)
+		server, err := NewCdiAPIServer("0.0.0.0", 0, client, aggregatorClient, cdiClient, authorizer, authConfigWatcher, nil)
+		Expect(err).ToNot(HaveOccurred())
 
-	if rr.Code != http.StatusOK {
-		t.Errorf("Unexpected status code %d", rr.Code)
-	}
-}
+		app := server.(*cdiAPIApp)
 
-func TestAuthConfigUpdate(t *testing.T) {
-	ch := make(chan struct{})
-	cm := getAPIServerConfigMap(t)
-	kubeobjects := []runtime.Object{}
-	kubeobjects = append(kubeobjects, cm)
-
-	client := k8sfake.NewSimpleClientset(kubeobjects...)
-	aggregatorClient := aggregatorapifake.NewSimpleClientset()
-	cdiClient := cdiclientfake.NewSimpleClientset()
-	authorizer := &testAuthorizer{}
-	acw := NewAuthConfigWatcher(client, ch).(*authConfigWatcher)
-
-	server, err := NewCdiAPIServer("0.0.0.0", 0, client, aggregatorClient, cdiClient, authorizer, acw, nil)
-	if err != nil {
-		t.Errorf("Upload api server creation failed: %+v", err)
-	}
+		req, _ := http.NewRequest("GET", "/apis", nil)
```

Comment:
I am assuming the missing result is the err, and we are expecting err to be nil, so we can do usual 'Expect(err).ToNot(HaveOccurred())'